gdal/ogr/ogrct.cpp

```diff
@@ -2399,6 +2442,37 @@ int OGRProjCT::TransformWithErrorCodes(
     return TRUE;
 }
 
+/************************************************************************/
+/*                               Clone()                                */
+/************************************************************************/
+
+OGRCoordinateTransformation* OGRProjCT::Clone() const override
+{
+    OGRProjCT* poNewCT = new OGRProjCT(*this);
+#if (PROJ_VERSION_MAJOR * 10000 + PROJ_VERSION_MINOR * 100 + PROJ_VERSION_PATCH) < 80001
+    // See https://github.com/OSGeo/PROJ/pull/2582
+    // This may fail before PROJ 8.0.1 if the m_pj object is a "meta"
+    // operation being a set of real operations
+    bool bCloneDone = ((m_pj == nullptr) == (poNewCT->m_pj == nullptr));
+    if(bCloneDone)
+    {
+        for(i = 0; i != m_oTransformations.size(); ++i)
+        {
+            bCloneDone = bCloneDone &&
+                         ((m_oTransformations[i].pj == nullptr) ==
+                          (poNewCT->m_oTransformations[i].pj == nullptr));
+        }
+    }
+    if(!bCloneDone)
+    {
+        delete poNewCT;
+        poNewCT = new OGRProjCT();
+        poNewCT->Initialize(poSRSSource, poSRSTarget, m_options);
```

Comment:
I'm not sure if I need to check the result of `Initialize` call.
I think, it is redundant.

Reply:
it should rarely fail, but that might be more prudent to check it